src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -229,6 +239,40 @@ RestExecuteOnExtensionResponse handleRestExecuteOnExtensionRequest(RestExecuteOn
         if (restHandler == null) {
             return new RestExecuteOnExtensionResponse("FAILED: No handler for " + restPath);
         }
+        Method m = null;
+        try {
+            m = restHandler.getClass().getMethod("handleRequest", RestRequest.Method.class, String.class);
+        } catch (NoSuchMethodException e) {
+            throw new RuntimeException("ExtensionRestHandler does not implement method handleRequest");
+        } catch (SecurityException e) {
+            throw new RuntimeException("ExtensionRestHandler does not implement method handleRequest");
+        }
+
+        if (m.isAnnotationPresent(RequiresPermissions.class)) {
+            Annotation annotation = m.getAnnotation(RequiresPermissions.class);
+            RequiresPermissions permissionsRequired = (RequiresPermissions) annotation;
+
+            // TODO validate that caller has the permissions required here
+            UUID principalUuid = request.getPrincipalUUID();
+
+            AuthorizationResponseHandler authorizationResponseHandler = new AuthorizationResponseHandler();
+            try {
+                AuthorizationResponse authResponse = this.extensionTransportService.submitRequest(
+                        opensearchNode,
+                        ExtensionsOrchestrator.REQUEST_EXTENSION_AUTHORIZE_ACTION,
+                        new AuthorizationRequest(principalUuid, Arrays.asList(permissionsRequired.permissions())),
+                        authorizationResponseHandler
+                ).get();
+            } catch (InterruptedException e) {
+                // Request should not be authorized if a definitive response cannot be received
+                throw new RuntimeException(e);
```

Comment:
not clear why we are throwing runtime exceptions rather than returning appropriate error codes to the user (timeout here, not authorized in the permissions case)